### PR TITLE
Solved Issue: Sidebar UI Enhancement

### PR DESCRIPTION
**Pull request by Monilprajapati**
I have raised this PR for the Issue #1254.

### CODE REVIEW

File: frontend/src/views/Docs.jsx [hunk 1]
```diff
@@ -98,29 +104,34 @@ const Docs = (defaultprops) => {
...
-		position: "sticky",
-		top: 50,
-		minHeight: "90vh",
-		maxHeight: "90vh",
-		overflowX: "hidden",
-		overflowY: "auto",
-		zIndex: 1000,
-		//borderRight: "1px solid rgba(255,255,255,0.3)",
+    minWidth: 300,
+    width: "20%",
```
Comment (frikky) [inline frontend/src/views/Docs.jsx:125]: Is this 20% of screen or 20% of parent?

**Review by frikky — COMMENTED** [review]
Your linter made it hard to see where you actually made code changes. Please point me to what you changed :)

File: frontend/src/views/Docs.jsx (hunk 1 above)
Comment (Monilprajapati) [reply frontend/src/views/Docs.jsx:125]: > Is this 20% of screen or 20% of parent?

Of parent which is Body which covers an entire screen i guess

**Comment by Monilprajapati** [comment]
> Your linter made it hard to see where you actually made code changes. Please point me to what you changed :)

Yeah sure

File: frontend/src/views/Docs.jsx [hunk 2]
```diff
@@ -19,11 +21,14 @@ import {
   Typography,
   Paper,
   List,
+  Collapse,
+  ListItemButton,
+  ListItemText
```
Comment (Monilprajapati) [inline frontend/src/views/Docs.jsx:26]: I have added this new material UI components for better UI

File: frontend/src/views/Docs.jsx [hunk 3]
```diff
@@ -57,27 +62,28 @@ const innerHrefStyle = {
...
+    //	return
+    //}
+  }, [])
+  //console.log("PARAMS: ", params)
 
   const [mobile, setMobile] = useState(serverMobile === true || isMobile === true ? true : false);
   const [data, setData] = useState("");
   const [firstrequest, setFirstrequest] = useState(true);
   const [list, setList] = useState([]);
+  const [isopen, setOpen] = useState(-1);
```
Comment (Monilprajapati) [inline frontend/src/views/Docs.jsx:86]: I have added this State to track the Collapse of an Individual item of sidebar

File: frontend/src/views/Docs.jsx [hunk 4]
```diff
@@ -98,29 +104,34 @@ const Docs = (defaultprops) => {
     setAnchorEl(null);
   }
 
+  const handleCollapse = (index) => {
+    setOpen(isopen === index ? -1 : index)
+  };
+
```
Comment (Monilprajapati) [inline frontend/src/views/Docs.jsx:110]: This is the function to handle the collapse like Open and close

File: frontend/src/views/Docs.jsx [hunk 5]
```diff
@@ -98,29 +104,34 @@ const Docs = (defaultprops) => {
...
   const SidebarPaperStyle = {
     backgroundColor: theme.palette.surfaceColor,
     overflowX: "hidden",
     position: "relative",
-    padding: 30,
+    paddingLeft: 15,
+    paddingRight: 15,
```
Comment (Monilprajapati) [inline frontend/src/views/Docs.jsx:116]: I have removed this Top padding for better UI

**Comment by Monilprajapati** [comment]
Hey @frikky I think this will work!

**Comment by Monilprajapati** [comment]
Hey @frikky Have you done with code review?

**Comment by frikky** [comment]
Cheers! Merging this in - nice and easy ;)

**Pull request merged**